src/dso_api/dynamic_api/permissions.py

```diff
@@ -111,6 +119,85 @@ def has_permission_for_models(self, request, view, models):
         log_access(request, access)
         return access
 
+    def _filters_ok(self, request, schema: DatasetTableSchema) -> bool:  # noqa: C901
+        """Check field authorization for requested filters."""
+        # Type hack: DRF requests have a __getattr__ that delegates to an HttpRequest.
+        request = cast(HttpRequest, request)
+
+        # And then we and authorization_django monkey-patch the scopes onto the request.
+        scopes = cast(UserScopes, request.user_scopes)
+
+        # Collect mandatory filters. We need to make an exception for these to handle cases
+        # where the field they reference isn't otherwise accessible.
+        # The actual check for whether the filter sets are provided lives elsewhere.
+        mandatory = set()
+        for profile in scopes.get_active_profile_tables(schema.dataset.id, schema.id):
+            for f in profile.mandatory_filtersets:
+                mandatory.update(f)
+
+        for key, values in request.GET.lists():
+            if key in ["fields", "sorteer"] or key.startswith("_"):
+                continue
+
+            # Everything else is a filter.
+            field_name, op = _parse_filter(key)
+
+            # mandatoryFilters may contain either the complete filter (with lookup operation)
+            # or just the field name.
+            if key in mandatory or field_name in mandatory:
+                continue
+
+            temporal = schema.temporal.dimensions if schema.temporal is not None else {}
+
+            if fields := temporal.get(field_name):
+                if not self._filter_ok(fields.start, scopes, schema):
+                    return False
+                if not self._filter_ok(fields.end, scopes, schema):
+                    return False
+
+            elif not self._filter_ok(field_name, scopes, schema):
+                return False
+
+        return True
+
+    def _filter_ok(self, field_name: str, scopes: UserScopes, schema: DatasetTableSchema) -> bool:
+        schema_part = schema
+        parts = field_name.split(".")  # Handle subfields.
+        for i, part in enumerate(parts):
+            try:
+                schema_part = schema_part.get_field_by_id(part)
+            except SchemaObjectNotFound as e:
+                # TODO: this should be a ValidationError,
+                # but that exception isn't handled properly from here.
+                # Raise PermissionDenied to at least get the exception message to the client.
+                raise PermissionDenied(f"{field_name} not found in schema") from e
```

Comment:
Ik zou verwachten dat een `raise ValidationError(..)` (wel de DRF class versie) prima werkt, aangezien de hele afhandeling binnen in de `APIView.dispatch()` valt. Permissie checks vinden plaats wanneer `request.user` wordt opgevraagd binnen `self.initial()`.

Reply:
Ik weet niet precies wat hier misging, maar dat is sowieso een ander ticket :)